### PR TITLE
checking the status code returned by youtube

### PR DESCRIPTION
**Pull request by unixfox**
Should provide a better error message for the 429 rate limit: https://github.com/iv-org/invidious/issues/5032

### CODE REVIEW

File: src/invidious/yt_backend/youtube_api.cr [hunk 1]
```diff
@@ -638,6 +638,11 @@ module YoutubeAPI
     # Send the POST request
     body = YT_POOL.client() do |client|
       client.post(url, headers: headers, body: data.to_json) do |response|
+        if response.status_code == 200
```
Comment (syeopite) [inline src/invidious/yt_backend/youtube_api.cr:641]:
```suggestion
        if response.status_code != 200
```

**Review by syeopite — CHANGES_REQUESTED** [review]

File: src/invidious/yt_backend/youtube_api.cr (hunk 1 above)
Comment (unixfox) [reply src/invidious/yt_backend/youtube_api.cr]: Sorry I left the wrong condition for testing the error message 😅

**Review by syeopite — APPROVED** [review]

**Review by SamantazFox — APPROVED** [review]

**Comment by unixfox** [comment]
I think based on the URL alone it's going to be pretty obvious.

Usually even on errors, the yt api just returns a 200 status code.

**Comment by iBicha** [comment]
> I think based on the URL alone it's going to be pretty obvious.
> 
> Usually even on errors, the yt api just returns a 200 status code.

I think it should at least contain the error from the body to make this more helpful than it was

**Comment by unixfox** [comment]
We would need an error template which summarize the error AND print the full error stack, but we don't have one.

For the moment, that's the best that we can do for clearly explaining to the user what's going on.

I still insist on the case that when you receive another status code other than 200 then something really wrong is going on. Many people will discover it and create new issues, the URL causing the issue will come up too at some point.

**Comment by iBicha** [comment]
I mean the YT response body can be just added to the body of InfoException, that is also fine

**Comment by unixfox** [comment]
But then these messages will never be able to be translated.

Hence why I said, we do not have a page template with a summary of the error AND the full stack error.

**Comment by unixfox** [comment]
And I forgot to say that we already parse the error given by youtube here: https://github.com/iv-org/invidious/blob/792d0d5f6df912039a58768e6ff503ae00abe7c0/src/invidious/yt_backend/youtube_api.cr#L654

Here https://github.com/iv-org/invidious/issues/5032 why Invidious is crashing. Is that it except youtube to give a JSON body but receive an HTML one.

Now that I think again. Maybe we could only throw a simple error status code message. When the status code is not 200 AND it's not a JSON body.

I think youtube probably return a JSON body with an error and a status code of 400

**Comment by syeopite** [comment]
I have a PR opened upstream in Kemal to allow for routes pertaining to specific raised errors https://github.com/kemalcr/kemal/pull/688

I haven't had time to work on it lately but once it is finished and merged we should be able to create customized error pages for different exceptions

**Pull request merged**